src/Features/CSharp/Portable/CodeFixes/GenerateMethod/GenerateDeconstructMethodCodeFixProvider.cs

```diff
@@ -95,6 +100,11 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
                 return;
             }
 
+            if (deconstruction is PositionalPatternClauseSyntax positionalPatternClause && positionalPatternClause.Subpatterns.Any(p => p.Pattern is not ConstantPatternSyntax))
```

Comment:
can you doc what thsi check is for?

Reply:
In the TryMakeParameters method we cast the Subpatterns of the positionalPattern to constant pattern, that way we can get down to the expression level of the positional pattern, which was found out through sharplab. So this check is here to make sure we can get to the expression syntax of the positional pattern through a constant pattern, if not we return.

When you say document, do you mean add it above the if statement as a simple comment? Sorry probably a stupid question but just making sure lol!

Reply:
Good to know! I will keep that in mind from now on! I definitely need to work on my comment game. Mostly use to working on my own projects.